src/libstore/build-result.hh

```diff
@@ -63,15 +64,26 @@ struct BuildResult
        non-determinism.) */
     bool isNonDeterministic = false;
 
+    /* For derivations, the derivation path and the wanted outputs. */
+    std::optional<StorePath> drvPath;
     DrvOutputs builtOutputs;
 
+    /* For substitutions, the substituted path. */
+    std::optional<StorePath> outPath;
+
```

Comment:
I don't really like this because it's possible for both to be `nullopt`. I was about to propose `std::variant<StorePath, StorePath>`, but why don't we just return the original `DerivedPath` used in the request? That's exactly correct!

Reply:
Thanks, done.